src/nvim/ex_cmds2.c

```diff
@@ -2788,6 +2788,43 @@ static FILE *fopen_noinh_readbin(char *filename)
   return fdopen(fd_tmp, READBIN);
 }
 
+typedef struct {
+  char_u *buf;
+  uint32_t pointer;
+} GetStrLineCookie;
+
+static char_u *get_str_line(int c, void *cookie, int ident)
+{
+  GetStrLineCookie *p = (GetStrLineCookie *)cookie;
```

Comment:
we never cast `void *` pointers (C doesn't require it, and it's a source of mistakes).